enhancements/cluster-logging/cluster-logging-log-forwarding.md

```diff
@@ -0,0 +1,310 @@
+
+
+
+### Non-Goals
+
+Log forward intends to be opinionated about the source of log messages; it will explicitly define what constitutes "application logs" or "infrastructure logs". It will not allow configuration of additional sources without further design considerations and how those sources may impact the collecting agent. Log forwarding is not intended to provide a complex routing solution as one might achieve by using a custom collector configuration or a messaging solution (e.g. kafka).
+
+## Proposal
+
+Log forwarding will provide a declarative way to specify the destination for specific types of logs using a 'pipeline'.  A 'pipeline' defines simple routing for one source to one or more destinations. The source of logs are opinionated and well defined by cluster logging. The initial source types are as follows:
+
+* `logs.app` - Container logs generated by user applications running on the platform, excluding infrastructure container applications
+* `logs.infra` - Logs generated by both infrastructure components running on the platform and OKD nodes (e.g. journal logs).  "Infra" applications are defined as any pods which run in namespaces: `openshift*`, `kube*`, `default`.
+
+Future types may be:
+* `events.kubernetes` - Kubernetes events
+* `audit.container-engine` - Container engine audit events
+
+There are no assumptions regarding whether or not an endpoint is deployed on or off cluster.  Endpoints off-cluster may require adminstrators to perform additional actions in order for logs to be forwarded (e.g. secret creation, opening port) 
+Following is the list of supported endpoint types for this proposal:
+
+* Elasticsearch
+* Fluent secure-forward
+
+### User Stories
+
+#### As an OKD admin, I want to deploy only a collector, and have it ship logs off-cluster to my own aggregation solution
+
+This is a typical example of organizations that desires to re-use their existing enterprise log solution.  We will succeeded if we are able to write logs to their logging service.
+
+#### As an OKD admin, I want to aggregate application logs on-cluster and infra logs off-cluster
+
+This is an example of an OKD cluster hosting solution where several organizations are each provided with a dedicated cluster.  The organization requires access to application container logs but the host requires access to the infra structure logs.
+
+### Implementation Details
+
+#### Assumptions
+
+* A pipeline shall support at least one target endpoint. 
+* A pipeline may support multiple target endpoints.
+* Endpoints that require secrets which are not created by the cluster-logging-operator shall be created and managed by the administrator of the endpoint
+* Secrets shall have keys of: cert,key,cacert which point to the respective certificates for which they represent
+
+#### Security
+* A pipeline may support TLS configuration by declaring a secret. Secrets must contain keys: cert, key, cacert
+* Endpoints defined without a secret will use unsecure communication
```

Comment:
It would be nice if here we wouldn't do "insecure by default"; meaning, if folks do need to use an insecure connection, then they should need to specify a flag such as: "insecure: true" or something along these lines. This makes it easier to catch such entries when doing a security audit or when attempting to harden the deployment.

Reply:
added to the api spec